utils/utils_test.go

```diff
@@ -188,3 +191,141 @@ func TestRandomNumber(t *testing.T) {
 		assert.Equal(t, 1, new(big.Int).Sub(maxAmount, result).Sign())
 	}
 }
+
+var (
+	blockIdentifier = &types.BlockIdentifier{
+		Hash:  "block",
+		Index: 1,
+	}
+
+	accountCoin = &types.AccountIdentifier{
+		Address: "test",
+	}
+
+	currency = &types.Currency{
+		Symbol:   "BLAH",
+		Decimals: 2,
+	}
+
+	amountCoins = &types.Amount{
+		Value:    "60",
+		Currency: currency,
+	}
+
+	accountCoins = []*types.Coin{
+		&types.Coin{
+			CoinIdentifier: &types.CoinIdentifier{Identifier: "coin1"},
+			Amount: &types.Amount{
+				Value:    "30",
+				Currency: currency,
+			},
+		},
+		&types.Coin{
+			CoinIdentifier: &types.CoinIdentifier{Identifier: "coin2"},
+			Amount: &types.Amount{
+				Value:    "30",
+				Currency: currency,
+			},
+		},
+	}
+
+	accountBalance = &types.AccountIdentifier{
+		Address: "test2",
+	}
+
+	amountBalance = &types.Amount{
+		Value:    "100",
+		Currency: currency,
+	}
+
+	accBalanceRequest1 = &AccountBalanceRequest{
+		Account:  accountCoin,
+		Currency: currency,
+	}
+
+	accBalanceResp1 = &AccountBalance{
+		Account: accountCoin,
+		Amount:  amountCoins,
+		Coins:   accountCoins,
+		Block:   blockIdentifier,
+	}
+
+	accBalanceRequest2 = &AccountBalanceRequest{
+		Account:  accountBalance,
+		Currency: currency,
+	}
+
+	accBalanceResp2 = &AccountBalance{
+		Account: accountBalance,
+		Amount:  amountBalance,
+		Block:   blockIdentifier,
+	}
+)
+
+func TestGetAccountBalances(t *testing.T) {
+	ctx := context.Background()
+	mockHelper := &MockFetcherHelper{}
+
+	accBalances, err := GetAccountBalances(
+		ctx,
+		mockHelper,
+		[]*AccountBalanceRequest{accBalanceRequest1, accBalanceRequest2},
+	)
+
+	assert.Nil(t, err)
+	assert.Equal(t, accBalances[0], accBalanceResp1)
+	assert.Equal(t, accBalances[1], accBalanceResp2)
+
+	// Returns error correctly
+	mockHelper.IsError = true
+	accBalances, err = GetAccountBalances(
+		ctx,
+		mockHelper,
+		[]*AccountBalanceRequest{accBalanceRequest1},
+	)
+	assert.Nil(t, accBalances)
+	assert.Error(t, err)
+}
+
+type MockFetcherHelper struct {
```

Comment:
nit: this isn't how you mock using the stuff you generated...take a look at for an example: https://github.com/coinbase/rosetta-sdk-go/blob/42a93b8550a614ac8d49627afb92ba6cb2774caf/syncer/syncer_test.go#L198-L222

You can remove everything from 290 down and import the mock you generated.